devops/Makefile

```diff
@@ -462,102 +396,48 @@ create-qa-api-db:
-
-create-ffa-api-db:
-	test -n $(NAMESPACE)
-	@echo "+\n++ Creating DB (FFA_API_DB) in $(NAMESPACE) patroni db cluster"
-	@oc -n $(NAMESPACE) process -f $(CREATE_FFA_DB_TEMPLATE_PATH) -p DB_NAME=$(FFA_API_DB_NAME) -p BUILD_NAMESPACE=$(BUILD_NAMESPACE) -p JOB_NAME=api | oc -n $(NAMESPACE) apply -f -
-
-deploy-forms-flow-secrets:
-	test -n "$(NAMESPACE)"
-	@oc -n $(NAMESPACE) process -f $(SECRETS_TEMPLATE_PATH) --param-file=$(SECRETS_PARAM_PATH) | oc -n $(NAMESPACE) apply -f -
-
-deploy-forms-flow-forms:
+deploy-forms:
 	test -n "$(NAMESPACE)"
+	test -n "$(FORMS_DEPLOY_TEMPLATE_PATH)"
+	test -n "$(FORMS_NAME)"
+	test -n "$(FORMS_URL)"
 	test -n "$(BUILD_NAMESPACE)"
+	test -n "$(BUILD_TAG)"
+	test -n "$(HOST_PREFIX)"
+	test -n "$(FORMIO_ROOT_EMAIL)"
+	test -n "$(MONGODB_URI)"
 	@echo "+\n++ Deploying forms-flow-forms with tag: $(BUILD_TAG)\n+"
-	@oc -n $(NAMESPACE) process -f $(FORM_AI_FORMS_DEPLOY_TEMPLATE_PATH) -p NAME=$(FORMSFLOW_FORMS_NAME) -p FFA_FORMS_URL=$(FORM_AI_FORMS_URL) -p TOOLS_NAMESPACE=$(BUILD_NAMESPACE) -p IMAGE_STREAM_TAG="$(BUILD_TAG)" | oc -n $(NAMESPACE) apply -f -
-	$(call rollout_and_wait,dc/$(FORMSFLOW_FORMS_NAME))
-
-deploy-forms-flow-bpm:
-	test -n "$(NAMESPACE)"
-	test -n "$(BUILD_NAMESPACE)"
-	@echo "+\n++ Deploying forms-flow-bpm with tag: $(BUILD_TAG)\n+"
-	@oc -n $(NAMESPACE) process -f $(FORM_AI_BPM_DEPLOY_TEMPLATE_PATH) -p NAME=$(FORMSFLOW_BPM_NAME) -p CAMUNDA_URL=$(FORM_AI_BPM_URL) -p TOOLS_NAMESPACE=$(BUILD_NAMESPACE) -p IMAGE_STREAM_TAG="$(BUILD_TAG)" | oc -n $(NAMESPACE) apply -f -
-	$(call rollout_and_wait,dc/$(FORMSFLOW_BPM_NAME))
-
-deploy-forms-flow-api:
-	test -n "$(NAMESPACE)"
-	test -n "$(BUILD_NAMESPACE)"
-	@echo "+\n++ Deploying forms-flow-api with tag: $(BUILD_TAG)\n+"
-	@oc -n $(NAMESPACE) process -f $(FORM_AI_API_DEPLOY_TEMPLATE_PATH) -p NAME=$(FORMSFLOW_API_NAME) -p WEBAPI_URL=$(FORM_AI_API_URL) -p TOOLS_NAMESPACE=$(BUILD_NAMESPACE) -p IMAGE_STREAM_TAG="$(BUILD_TAG)" | oc -n $(NAMESPACE) apply -f -
-	$(call rollout_and_wait,dc/$(FORMSFLOW_API_NAME))
-
-deploy-forms-flow-web:
-	test -n "$(NAMESPACE)"
-	test -n "$(BUILD_NAMESPACE)"
-	@echo "+\n++ Deploying forms-flow-web with tag: $(BUILD_TAG)\n+"
-	@oc -n $(NAMESPACE) process -f $(FORM_AI_WEB_DEPLOY_TEMPLATE_PATH) --param-file=$(WEB_PARAM_PATH) -p NAME=$(FORMSFLOW_WEB_NAME) -p FFAWEB_URL=$(FORM_AI_WEB_URL) -p TOOLS_NAMESPACE=$(BUILD_NAMESPACE) -p IMAGE_STREAM_TAG="$(BUILD_TAG)" | oc -n $(NAMESPACE) apply -f -
-	$(call rollout_and_wait,dc/$(FORMSFLOW_WEB_NAME))
+	@oc -n $(NAMESPACE) process -f $(FORMS_DEPLOY_TEMPLATE_PATH) \
+	-p NAME=$(FORMS_NAME) \
+	-p FORMS_URL=$(FORMS_URL) \
+	-p TOOLS_NAMESPACE=$(BUILD_NAMESPACE) \
+	-p IMAGE_STREAM_TAG=$(BUILD_TAG) \
+	-p HOST_PREFIX=$(HOST_PREFIX) \
+	-p FORMIO_ROOT_EMAIL=$(FORMIO_ROOT_EMAIL) \
+	-p MONGODB_URI=$(MONGODB_URI) \
+	| oc -n $(NAMESPACE) apply -f -
+	$(call rollout_and_wait,dc/$(HOST_PREFIX)$(FORMS_NAME))
```

Comment:
This concatenation `$(HOST_PREFIX)$(FORMS_NAME)` happens differently for the others. Any reason to do it differently here?
- API: $(call rollout_and_wait,dc/$(API))
- Workers: $(call rollout_and_wait,dc/$(WORKERS))
- Queue consumers: $(call rollout_and_wait,dc/$(QUEUE_CONSUMERS))
- Web: $(call rollout_and_wait,dc/$(WEB))

Reply:
call rollout_and_wait just checks if the particular deployment config is up or not, with the app label name. For the others the name has the hostprefix concatenation already done when the github action is triggered, if its called directly the api will be created without the hostprefix. In this implementation wherever its called, the hostprefix is concatenated and the call rollout is waiting for the particular name with hostprefix.

Reply:
I know that 😉 
My point was "happens differently for the others". Why not follow what the others are doing or change the others to follow this or whatever is needed to make them all the same?